tests/test_cli.py

```diff
@@ -188,11 +188,19 @@ def test_cli_connection_tests_success(caplog, functional_engine, runner):
     assert "Successfully connected to the Symplectic Elements FTP server" in caplog.text
 
 
-def test_cli_connection_tests_fail(
-    caplog, ftp_server, monkeypatch, nonfunctional_engine, runner
+def test_cli_database_connection_test_fails(caplog, nonfunctional_engine, runner):
+    with patch("carbon.cli.DatabaseEngine") as mocked_engine:
+        mocked_engine.return_value = nonfunctional_engine
+        result = runner.invoke(main, ["--run_connection_tests"])
+        assert result.exit_code == 1
+
+    assert "Failed to connect to the Data Warehouse" in caplog.text
```

Comment:
To me, this is a nice test.  The fixture `nonfunctional_engine` is clearly worded, indicating that it's doomed from the get-go, and the test confirms it.